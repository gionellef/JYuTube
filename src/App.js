import React from 'react';

import { Grid } from '@material-ui/core';

import { SearchBar, VideoDetails, VideoList } from "./components";

import youtube from './api/youtube';

// Components can either be class based components or function-based / dummy components
class App extends React.Component {
  state = {
    videos: [],
    selectedVideo: null,
  }

  // will be run immediately after class component loads
  componentDidMount() {
    this.handleSubmit();
  }

  onVideoSelect = (video) => {
    this.setState({ selectedVideo: video });
  }

  handleSubmit = async (searchTerm) => {
    const response = await youtube.get('search',{
      params: {
        part: 'snippet',
        maxResults: 5,
        key: 'AIzaSyDiLzOrZe5yYpGoslPMANLrspnkGoJSyPs',
        q: searchTerm
      }
    });
    
    this.setState({ videos: response.data.items, selectedVideo: response.data.items[0] });
  }

  render () {
    const { selectedVideo, videos } = this.state;
    return (
      <Grid justify="center" style={{ width: '100vw' }} container spacing={10}>
        <Grid item xs={12}>
          <Grid container spacing={10}>
            <Grid item xs={12}>
              <SearchBar onFormSubmit={this.handleSubmit} />
            </Grid>
            <Grid item xs={8}>
              <VideoDetails video={selectedVideo} />
            </Grid>
            <Grid item xs={4}>
              <VideoList videos={videos} onVideoSelect={this.onVideoSelect} />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    )
  }
}

export default App;