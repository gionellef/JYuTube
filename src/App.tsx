import * as React from 'react';

import AppLayout from './components/AppLayout/AppLayout';
import { Home } from './containers/Home/Home';
import { Watch } from './containers/Watch/Watch';
import { Route, Switch } from 'react-router-dom';


class App extends React.Component<any, any> {
  render () {
    return (
      <AppLayout>
        <Switch>
          <Route path='/watch' component={ Watch } />
          <Route path='/' exact component={ Home } />
        </Switch>
      </AppLayout>
    );
  }
}

export default App;