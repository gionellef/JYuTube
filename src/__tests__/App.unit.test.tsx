import * as React from 'react';
import App from '../App';
import { shallow } from 'enzyme';

describe('App', () => {
  test('renders App', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});