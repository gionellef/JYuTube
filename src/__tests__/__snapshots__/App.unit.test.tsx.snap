// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
<AppLayout>
  <Switch>
    <Route
      component={[Function]}
      path="/watch"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
  </Switch>
</AppLayout>
`;
